pkg/controller/controller.go

```diff
@@ -337,6 +337,18 @@ func (c *controller) reconcileServiceClassFromBrokerCatalog(broker *v1alpha1.Bro
 		return err
 	}
 
+	if existingServiceClass.BrokerName != broker.Name {
```

Comment:
Can we get some hypotheticals in a comment on how these conditions should happen?